src/Quantum/MP.jl

```diff
@@ -110,6 +111,17 @@ end
 const MPS = MatrixProduct{State}
 const MPO = MatrixProduct{Operator}
 
+tensors(ψ::TensorNetwork{MatrixProduct{P,Infinite}}, args...) where {P<:Plug} =
+    throw(MethodError("You need to specify the site for an infinite MatrixProduct$P"))
+
```

Comment:
I meant don't even put a method jajaja. Sorry if I didn't explain it well.

The Julia Runtime will already throw a `MethodError` if a matching method doesn't exist.

```suggestion
```